src/backend/catalog/system_functions.sql

```diff
@@ -722,6 +722,76 @@ REVOKE EXECUTE ON FUNCTION pg_ls_logicalmapdir() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_wal_lsn_diff(pg_lsn, pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_is_in_recovery() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_last_wal_receive_lsn() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_last_wal_replay_lsn() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_last_xact_replay_timestamp() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_is_wal_replay_paused() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_get_wal_replay_pause_state() FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_copy_physical_replication_slot(src_slot_name name, dst_slot_name name, temporary boolean, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_copy_physical_replication_slot(src_slot_name name, dst_slot_name name, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_drop_replication_slot(name) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_copy_logical_replication_slot(src_slot_name name, dst_slot_name name, temporary boolean, plugin name, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_copy_logical_replication_slot(src_slot_name name, dst_slot_name name, temporary boolean, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_copy_logical_replication_slot(src_slot_name name, dst_slot_name name, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_slot_peek_changes(slot_name name, upto_lsn pg_lsn, upto_nchanges integer, VARIADIC options text[], OUT lsn pg_lsn, OUT xid xid, OUT data text) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_slot_peek_binary_changes(slot_name name, upto_lsn pg_lsn, upto_nchanges integer, VARIADIC options text[], OUT lsn pg_lsn, OUT xid xid, OUT data bytea) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_replication_slot_advance(slot_name name, upto_lsn pg_lsn, OUT slot_name name, OUT end_lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_emit_message(boolean, text, text) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_emit_message(boolean, text, bytea) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_create_logical_replication_slot(slot_name name, plugin name, temporary boolean, twophase boolean, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_terminate_backend(pid integer, timeout bigint) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_slot_get_changes(slot_name name, upto_lsn pg_lsn, upto_nchanges integer, VARIADIC options text[], OUT lsn pg_lsn, OUT xid xid, OUT data text) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_logical_slot_get_binary_changes(slot_name name, upto_lsn pg_lsn, upto_nchanges integer, VARIADIC options text[], OUT lsn pg_lsn, OUT xid xid, OUT data bytea) FROM PUBLIC;
+
+REVOKE EXECUTE ON FUNCTION pg_create_physical_replication_slot(slot_name name, immediately_reserve boolean, temporary boolean, OUT slot_name name, OUT lsn pg_lsn) FROM PUBLIC;
+
```

Comment:
I don't think we should want to include this in our own PG version - it'll break current users' workflows and (permanently!) remove features from users' databases.

(Permanently, until we manually fix it using an system admin account)